### ISSUE TITLE
How to make stuff drop bags?

### ISSUE THREAD

**Issue by firespark81** [DoubleDoorDevelopment/JsonLootBags#4]
How do you make things drop bags when all bags are the same? I am trying to use Quadrum to have a block drop a bag when I break it but all I get is an invalid bag when I use "JsonLootBags:ItemLootBag". So how do you use this mod with others when you can specify a type such a JsonLootBags:ItemLootBag.withTag({type: "TreasureChest"})?

**Comment by dries007** [comment]
Same thing, you add `type: "<name here>"` NBT tag to the item.
!BUT!
The types are not just the same as the vanilla types of loot chests.
You can add, change, and remove types, by looking in the `config/JsonLootBags`
Any .json files in there will be loaded in as types. If you want to get the build in defaults back, delete the folder and restart your client/server.

**Comment by firespark81** [comment]
Quadrum does not have an option to add a type and when I do I still get the invalid bag. Also I dont want the defaults I want my custom bag to drop. I get the json part and how it all works I just cant figure out how to get it to drop when most other custom this/that mods dont have an option for a type. They all just want modname:block and meta. The author should really consider giving each bag its own name example JsonLootBags:mybag. Because even if I just put JsonLootBags:ItemLootBag you would thing you would get a random bag. But you dont.

**Comment by dries007** [comment]
(I am the author btw)
Why would Quadrum have no support for NBT date? Its used by many, many things.
Making one item per type would defeat the purpose of making it configurable, as you can't just add and remove items when you switch servers.
Using metadata would result in issues too, as it would break on any configuration change.
NBT data is the only way that makes sense.

**Comment by firespark81** [comment]
I found a mod that assigns meta to bags when they are made using a config similar to yours and it looks like you are right because the bags just crash when you try to open them. It appears I'm not the only one with that issue with that mod as well. Also found the code for Quadrums' block drop input and confirmed that he does not support NBT data. http://pastebin.com/wknPyE25 It appears all other mods I have attempted to use have the same problem. Seems no one uses meta data support, which sucks cause your mod is so freaking awesome and I am determined to use it. 

Just an idea but what if in the future you had it so that when you create a JSON for a bag it actually makes a new item similar to how you create new items with say Quadrum or MetaMod. When you create a JSON in these mods the item gets assigned a name that you put in the JSON and that name is attached to the ModID. So then when someone creates a new bag it would be JsonLootBags:Bagname. This would allow you to leave out the NBT data and make the mod more compatible with other mods. I could see modpack makers doing a lot of really cool things with this mod if it had this ability. 

Also after re-reading my last post I feel it may have sounded a little harsh. I apologize for that. I wrote it first thing in the morning and I wasn't fully awake. This is a really great mod and thank you for all your hard work.

**Comment by dries007** [comment]
Well, I feel like adding NBT support to Quadrums's drops would be the easiest solution.
I'll look into it myself if the author doesn't want too.

**Comment by firespark81** [comment]
I'm sure you are right and I messaged him but have not heard back. I only found two custom things mods that support NBT and that is minetweaker and Block Properties. I however could not figure out how to use Block Properties. Seems to be a very complex mod and while he has a very detailed wiki for his commands he gives no examples. Anyway thank you so much for the fast replies and all the help/info. And Thanks again for the great mod. :)
